src/mvt_getter.py

```diff
@@ -0,0 +1,297 @@
+# -*- coding: utf-8 -*-
+import json
+from mapcss import MapCSS
+import os
+import sys
+import math
+from optparse import OptionParser
+
+reload(sys)
+sys.setdefaultencoding("utf-8")  # a hack to support UTF-8
+
+
+def pixel_size_at_zoom(z, l=1):
+    """
+    Converts l pixels on tiles into length on zoom z
+    """
+    return int(math.ceil(l * 20037508.342789244 / 512 * 2 / (2 ** z)))
+
+
+def get_vectors(minzoom, maxzoom, x, y, style, vec):
+    geomcolumn = "way"
+
+    pxtolerance = 1.8
+    table = {
+        "polygon": "planet_osm_polygon",
+        "line": "planet_osm_line",
+        "point": "planet_osm_point",
+    }
+
+    types = {"line": "line", "polygon": "area", "point": "node"}
+
+    names = set()
+    adp = ""
+
+    if "get_sql_hints" in dir(style):
+        adp = []
+
+        for zoom in range(minzoom, maxzoom):
+            names.update(style.get_interesting_tags(types[vec], zoom))
+
+            sql_hint = style.get_sql_hints(types[vec], zoom)
+            for tp in sql_hint:
+                add = []
+                for j in tp[0]:
+                    if j not in names:
+                        break
+                else:
+                    add.append(tp[1])
+                if add:
+                    add = " OR ".join(add)
+                    add = "(" + add + ")"
+                    adp.append(add)
+
+    adp = " OR ".join(adp)
+    if adp:
+        adp = adp.replace("&lt;", "<")
+        adp = adp.replace("&gt;", ">")
+
+    select = ",".join(["tags->" + "'" + i + "'" + ' as "' + i + '"' for i in names])
+    groupby = ",".join(['"' + i + '"' for i in names])
+
+    if vec == "polygon":
+        query = """select ST_AsMVTGeom(w.way, ST_TileEnvelope(%s, %s, %s), 4096, 64, true) as %s, %s from
```

Comment:
for last zoom we may want higher Extent. if we stop at zoom 14 and want map until zoom 22, each pixel should have 2**(22-14)=256 gradations inside = 512*256 = 131072? at least 8192 will get a pixel perfect zoom 18.